### ISSUE TITLE
_this.XHRType() of undefined"

### ISSUE THREAD

**Issue by SivaJonnala** [paulvanbladel/aurelia-auth#64]
Hi , I am using Aurelia_Auth, In that I am called Login(email, password). I got the error _this.XHRType() of undefined; I did the debugging process I got the error at this moment..var xhr = _this.xhr = new _this.XHRType(); Is there any way to resolve this error? "

**Comment by RWOverdijk** [comment]
@SivaJonnala Can you show me your code? Because this is standard functionality that works, so I'm curious how you're breaking it.

**Comment by SivaJonnala** [comment]
@RWOverdijk Hi, I provided my code at this.
https://gist.github.com/SivaJonnala/cbdf2e7e773bb8eec01c
![image](https://cloud.githubusercontent.com/assets/10843991/11742723/2384f68e-a028-11e5-9369-88e37630a962.png)
Please look at the code once and I got the error at this moment of the pic

**Comment by RWOverdijk** [comment]
@SivaJonnala Sorry for the delay in answering. You're using the http-client? I recommend using the fetch client (also because I have more experience with that lib so it's easier to help). I have support for it in [my fork](https://github.com/spoonx/aurelia-auth) of aurelia auth. I'm also shifting my support for aurelia-auth over there

In can't see a whole lot wrong here to be honest. It has been 8 days, have you found anything yet? If not, we can take a deeper dive into this.

**Comment by SivaJonnala** [comment]
@RWOverdijk Hi, I have found a solution for that, yes you said the right thing using http-client. it is giving errors, with the login method I got the token, but the new problem I got is, Aurelia_auth is not sending the token for every network calls. So I have added the token explicitly to the web service calling. But one thing I realized is, If the aurelia_auth does not able to send the token, then the better functionality of Aurelia_auth got failed. And I am very much thankful to you for your responsiveness. I will see your fork also I hope it will give me good expected results.
